pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -66,23 +76,56 @@ func ApplyMonitoringConsole(client splcommon.ControllerClient, cr splcommon.Meta
 		return err
 	}
 
-	statefulsetMC, err := getMonitoringConsoleStatefulSet(client, cr, &spec, SplunkMonitoringConsole, secretName)
-	if err != nil {
-		return err
+	//by default set multisite to false
+	multiSite := "false"
+
+	//get cluster info from cluster master
+	if cr.GetObjectKind().GroupVersionKind().Kind == "ClusterMaster" && !spec.Mock {
+		mgr := monitoringConsolePodManager{cr: &cr, spec: &spec, secrets: secrets, newSplunkClient: splclient.NewSplunkClient}
+		c := mgr.getClusterMasterClient(cr)
+		clusterInfo, err := c.GetClusterInfo(spec.Mock)
+		if err != nil {
+			return err
+		}
+		multiSite = clusterInfo.MultiSite
 	}
 
-	mgr := splctrl.DefaultStatefulSetPodManager{}
-	_, err = mgr.Update(client, statefulsetMC, 1)
+	statefulset, err := getMonitoringConsoleStatefulSet(client, cr, &spec, SplunkMonitoringConsole, secretName, multiSite)
 	if err != nil {
 		return err
 	}
 
+	mgr := splctrl.DefaultStatefulSetPodManager{}
+	_, err = mgr.Update(client, statefulset, 1)
+
 	return err
 }
 
+// getMonitoringConsoleClient for monitoringConsolePodManager returns a SplunkClient for monitoring console
+func (mgr *monitoringConsolePodManager) getMonitoringConsoleClient(cr splcommon.MetaObject) *splclient.SplunkClient {
+	fqdnName := splcommon.GetServiceFQDN(cr.GetNamespace(), GetSplunkServiceName(SplunkMonitoringConsole, cr.GetNamespace(), false))
+	return mgr.newSplunkClient(fmt.Sprintf("https://%s:8089", fqdnName), "admin", string(mgr.secrets.Data["password"]))
+}
+
+// getClusterMasterClient for monitoringConsolePodManager returns a SplunkClient for cluster master
+func (mgr *monitoringConsolePodManager) getClusterMasterClient(cr splcommon.MetaObject) *splclient.SplunkClient {
```

Comment:
I think both these APIs, `getClusterMasterClient` and `getMonitoringConsoleClient`, can be a wrapper around a common API( like for e.g.`getSplunkClient`) which takes instanceType as one of the arguments and returns the `splclient.SplunkClient`.